qcodes/instrument_drivers/Keysight/private/Keysight_344xxA.py

```diff
@@ -0,0 +1,455 @@
+
+        self._instrument.display_clear()
+
+        return numvals
+
+
+class _Keysight_344xxA(VisaInstrument):
+    """
+    Instrument class for Keysight 34460A, 34461A, 34465A and 34470A multimeters.
+
+    Tested with: 34461A, 34465A.
+
+    The driver currently only supports using the instrument as a voltmeter.
+
+    Attributes:
+        model (str): The model number of the instrument
+        NPLC_list (list): A list of the available Power Line Cycle settings
+        ranges (list): A list of the available voltage ranges
+    """
+
+    def __init__(self, name, address, silent=False,
+                 **kwargs):
+        """
+        Create an instance of the instrument.
+
+        Args:
+            name (str): Name used by QCoDeS. Appears in the DataSet
+            address (str): Visa-resolvable instrument address.
+            silent (bool): If True, the connect_message of the instrument
+                is supressed. Default: False
+        Returns:
+            _Keysight_344xxA
+        """
+
+        super().__init__(name, address, terminator='\n', **kwargs)
+
+        idn = self.IDN.get()
+        self.model = idn['model']
+
+        DIG = 'DIG' in self._licenses()
+
+        ####################################
+        # Instrument specifications
+
+        PLCs = {'34460A': [0.02, 0.2, 1, 10, 100],
+                '34461A': [0.02, 0.2, 1, 10, 100],
+                '34465A': [0.02, 0.06, 0.2, 1, 10, 100],
+                '34470A': [0.02, 0.06, 0.2, 1, 10, 100]
+                }
+        if DIG:
+            PLCs['34465A'] = [0.001, 0.002, 0.006] + PLCs['34465A']
+            PLCs['34470A'] = [0.001, 0.002, 0.006] + PLCs['34470A']
+
+        ranges = {'34460A': [10**n for n in range(-3, 9)],  # 1 m to 100 M
+                  '34461A': [10**n for n in range(-3, 9)],  # 1 m to 100 M
+                  '34465A': [10**n for n in range(-3, 10)],  # 1 m to 1 G
+                  '34470A': [10**n for n in range(-3, 10)],  # 1 m to 1 G
+                  }
+
+        # The resolution factor order matches the order of PLCs
+        res_factors = {'34460A': [300e-6, 100e-6, 30e-6, 10e-6, 3e-6],
+                       '34461A': [100e-6, 10e-6, 3e-6, 1e-6, 0.3e-6],
+                       '34465A': [3e-6, 1.5e-6, 0.7e-6, 0.3e-6, 0.1e-6,
+                                  0.03e-6],
+                       '34470A': [1e-6, 0.5e-6, 0.3e-6, 0.1e-6, 0.03e-6,
+                                  0.01e-6]
+                       }
+        if DIG:
+            res_factors['34465A'] = [30e-6, 15e-6, 6e-6] + res_factors['34465A']
+            res_factors['34470A'] = [30e-6, 10e-6, 3e-6] + res_factors['34470A']
+
+        self._resolution_factors = res_factors[self.model]
+        self.ranges = ranges[self.model]
+        self.NPLC_list = PLCs[self.model]
+
+        if DIG:
+            self._apt_times = apt_times[self.model]
+
+        ####################################
+        # PARAMETERS
+
+        self.add_parameter('line_frequency',
```

Comment:
Shall the values of this parameter be limited to `vals=vals.Enum(50, 60)`?

Reply:
I don't think, it has only a get, you can't set it. Do we need validator in this case?

Reply:
oh, no, but then please add `set_cmd=False` to the `add_parameter` call - this way you show that setting is indeed not possible. You could also add a `dosctring=` argument where you explain that "the instrument returns the frequency of the grid that it is plugged into" or smth.

Reply:
btw, sorry for misinformation, set_cmd is False by default, so no need for explicitly adding it.

Reply:
True, I added it.

Reply:
Something else for merging?